src/regex.nim

```diff
@@ -2100,6 +2101,45 @@ proc groupsCount*(m: RegexMatch): int =
   ##
   m.groups.len
 
+
+proc groupNames*(m: RegexMatch): seq[string] =
+  ## return the names of capturing groups
+  ##
+  ## .. code-block:: nim
+  ##   block:
+  ##     let text = "hello world"
+  ##     var m: RegexMatch
+  ##     doAssert text.match(re"(?P<greet>hello) (?P<who>world)", m)
+  ##     doAssert m.groupCapture("greet", text) == "hello"
+  ##     doAssert m.groupCapture("who", text) == "world"
+
+  result = newSeq[string]()
+  result = toSeq(m.namedGroups.keys)
+
+
+proc groupCapture*(m: RegexMatch, groupName: string, text: string): string =
+  ##  Return last capture for a given capturing group
+  ##
+  ## .. code-block:: nim
+  ##   block:
+  ##  
+  ##     let text = "hello world her"
+  ##     var m: RegexMatch
+  ##     doAssert text.match(re"(?P<greet>hello) (?P<who>world) (?P<who>her)", m)
+  ##     doAssert m.groupsCount == 3
+  ##  
+  ##     for group in @["greet", "who"]:
+  ##       doAssert m.groupNames.contains(group)
+  ##  
```

Comment:
I think the example should just contain a `groupCapture` example, no need to show how to use groupCount and groupNames